pkg/tests/tasks/security/authorization/tcp_test.go

```diff
@@ -16,156 +16,195 @@ package authorization
-			if !strings.Contains(msg, "connection succeeded") {
-				log.Log.Errorf("Verify setup Unexpected response: %s", msg)
-				t.Errorf("Verify setup Unexpected response: %s", msg)
-			} else {
-				log.Log.Infof("Success. Get expected response: %s", msg)
-			}
-		} else {
-			tcpEchoPod, _ := util.GetPodName("foo", "app=tcp-echo")
-			podIP, _ := util.Shell(`kubectl get pod %s -n foo -o jsonpath="{.status.podIP}"`, tcpEchoPod)
-			cmd := fmt.Sprintf(`sh -c 'echo "port %s" | nc %s %s' | grep "hello" && echo 'connection succeeded' || echo 'connection rejected'`, port, podIP, port)
-			msg, err := util.PodExec("foo", sleepPod, "sleep", cmd, true)
-			util.Inspect(err, "Failed to get response", "", t)
-			if !strings.Contains(msg, "connection succeeded") {
-				log.Log.Errorf("Verify setup Unexpected response: %s", msg)
-				t.Errorf("Verify setup Unexpected response: %s", msg)
-			} else {
-				log.Log.Infof("Success. Get expected response: %s", msg)
-			}
-		}
-	}
-
-	t.Run("Security_authorization_rbac_allow_GET_tcp", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Apply a policy to allow requests to port 9000 and 9001")
-		util.KubeApplyContents("foo", TCPAllowPolicy)
-		time.Sleep(time.Duration(10) * time.Second)
-
-		ports := []string{"9000", "9001", "9002"}
-		for _, port := range ports {
-			if port == "9000" || port == "9001" {
-				cmd := fmt.Sprintf(`sh -c 'echo "port %s" | nc tcp-echo %s' | grep "hello" && echo 'connection succeeded' || echo 'connection rejected'`, port, port)
-				msg, err := util.PodExec("foo", sleepPod, "sleep", cmd, true)
-				util.Inspect(err, "Failed to get response", "", t)
-				if !strings.Contains(msg, "connection succeeded") {
-					log.Log.Errorf("Verify allow GET Unexpected response: %s", msg)
-					t.Errorf("Verify allow GET Unexpected response: %s", msg)
-				} else {
-					log.Log.Infof("Success. Get expected response: %s", msg)
-				}
-			} else {
-				tcpEchoPod, _ := util.GetPodName("foo", "app=tcp-echo")
-				podIP, _ := util.Shell(`kubectl get pod %s -n foo -o jsonpath="{.status.podIP}"`, tcpEchoPod)
-				cmd := fmt.Sprintf(`sh -c 'echo "port %s" | nc %s %s' | grep "hello" && echo 'connection succeeded' || echo 'connection rejected'`, port, podIP, port)
-				msg, err := util.PodExec("foo", sleepPod, "sleep", cmd, true)
-				util.Inspect(err, "Failed to get response", "", t)
-				if !strings.Contains(msg, "connection rejected") {
-					log.Log.Errorf("Verify allow GET Unexpected response: %s", msg)
-					t.Errorf("Verify allow GET Unexpected response: %s", msg)
-				} else {
-					log.Log.Infof("Success. Get expected response: %s", msg)
-				}
-			}
-		}
-	})
+func assertPortTcpEchoAccepted(t test.TestHelper, ns string, port string) {
+	cmd := fmt.Sprintf(`sh -c 'echo "port %s" | nc tcp-echo %s' | grep "hello" && echo 'connection succeeded' || echo 'connection rejected'`,
+		port, port)
+	oc.Exec(t,
+		pod.MatchingSelector("app=sleep", ns),
+		"sleep",
+		cmd,
```

Comment:
cmd can be inlined directly with 
```fmt.Sprintf(`sh -c 'echo "port %s" | nc tcp-echo %s' | grep "hello" && echo 'connection succeeded' || echo 'connection rejected'`,port, port)```

Reply:
This breaks the test and it failed to run that command. 
If we split that line to multiple lines of a string. I am getting the following error.

>  tcp_test.go:102: transient error (will retry): Command failed: "kubectl exec -n foo sleep-c7b98cf9d-jwf4g -c sleep -- sh -c 'echo \"port 9000\" | nc tcp-echo 9000' | grep \"hello\" && echo 'connection succeeded' \n\t\t|| echo 'connection rejected'"
        Error: command failed: hello port 9000
        connection succeeded
        sh: -c: line 1: syntax error near unexpected token `||'
        sh: -c: line 1: `		|| echo 'connection rejected''
    tcp_test.go:47: Attempt 1/30 failed. Retrying...

Reply:
I can remove the cmd and put the commands in the oc.Exec. I will update that.

Reply:
Ok, but seems to be failing because a syntax error, maybe I put something wrong in my example